src/Pos/Block/Pure.hs

```diff
@@ -584,4 +625,4 @@ verifyBlocks curSlotId verifyNoUnknown bvd initLeaders = view _3 . foldl' step s
                 , vbpMaxSize = Just (bvdMaxBlockSize bvd)
                 , vbpVerifyNoUnknown = verifyNoUnknown
                 }
-        in (newLeaders, Just $ getBlockHeader blk, res <> verifyBlock vbp blk)
+        in (newLeaders, newPsks, Just $ getBlockHeader blk, res <> verifyBlock vbp blk)
```

Comment:
`initPsks` is the initial set of certificates valid relatively to the beginning of blocks list. We iterate as follows:
1. Check that block is valid due to `psks`
2. Upgrade `psks` with a dlg data in block body
3. Switch to the next block